src/js/components/Carousel/Carousel.js

```diff
@@ -12,199 +12,169 @@ import { Keyboard } from '../Keyboard';
-    const { play } = this.props;
-    if (play && (!prevProps.play || play !== prevProps.play)) {
-      this.play();
-    } else if (!play && prevProps.play) {
-      clearInterval(this.timer);
-    }
-  }
-
-  componentWillUnmount() {
-    clearInterval(this.timer);
-  }
-
-  play = () => {
-    const { play } = this.props;
-    clearInterval(this.timer);
-    this.timer = setInterval(() => {
-      const { children } = this.props;
-      const { activeIndex } = this.state;
+  const play = () => {
+    const { children } = props;
+    clearInterval(timer);
+    timer = setInterval(() => {
       const lastIndex = Children.count(children) - 1;
       if (activeIndex < lastIndex) {
-        this.setState({
-          activeIndex: activeIndex + 1,
-          priorActiveIndex: activeIndex,
-        });
+        setPriorActiveIndex(activeIndex);
+        setActiveIndex(activeIndex + 1);
       } else {
-        this.setState({ activeIndex: 0, priorActiveIndex: activeIndex });
+        setPriorActiveIndex(activeIndex);
+        setActiveIndex(0);
       }
-    }, play);
+    }, playProps);
   };
 
-  onRight = () => {
-    const { activeIndex } = this.state;
-    clearInterval(this.timer);
-    this.setState({
-      activeIndex: activeIndex + 1,
-      priorActiveIndex: activeIndex,
-    });
+  useEffect(() => {
+    if (playProps) {
+      play();
+    } else {
+      clearInterval(timer);
+    }
+    return () => {
+      clearInterval(timer);
+    };
+  }, [playProps]);
+
+  const onRight = () => {
+    clearInterval(timer);
+    setPriorActiveIndex(activeIndex);
+    setActiveIndex(activeIndex + 1);
   };
 
-  onLeft = () => {
-    const { activeIndex } = this.state;
-    clearInterval(this.timer);
-    this.setState({
-      activeIndex: activeIndex - 1,
-      priorActiveIndex: activeIndex,
-    });
+  const onLeft = () => {
+    clearInterval(timer);
+    setPriorActiveIndex(activeIndex);
+    setActiveIndex(activeIndex - 1);
   };
 
-  onSelect = index => () => {
-    const { activeIndex } = this.state;
-    clearInterval(this.timer);
-    this.setState({ activeIndex: index, priorActiveIndex: activeIndex });
+  const onSelect = index => () => {
+    clearInterval(timer);
+    setPriorActiveIndex(activeIndex);
+    setActiveIndex(index);
   };
 
-  render() {
-    const { children, controls, fill, focus, theme, ...rest } = this.props;
-    const { activeIndex, priorActiveIndex } = this.state;
-
-    const showArrows = controls && controls !== 'selectors';
-    const showSelectors = controls && controls !== 'arrows';
-    const lastIndex = Children.count(children) - 1;
-    const onLeft = activeIndex > 0 ? this.onLeft : undefined;
-    const onRight = activeIndex < lastIndex ? this.onRight : undefined;
-
-    const CurrentIcon = theme.carousel.icons.current;
-    const iconColor = normalizeColor(
-      theme.carousel.icons.color || 'control',
-      theme,
+  const { children, controls, fill, focus, theme, ...rest } = props;
```

Comment:
can we move this destructure as well into the function definition?

Reply:
yes, will do